arch/src/x86_64/mod.rs

```diff
@@ -675,6 +675,25 @@ pub fn generate_common_cpuid(
         });
     }
 
+    // Check if KVM provides 0x8000_0006 (L2 cache information - added Linux 5.7) else
+    // fetch host CPUID version if available. This is not as good as the KVM version
+    // due to handling of hybrid CPUs.
+    if !cpuid.iter().any(|c| c.function == 0x8000_0006)
+        // SAFETY: cpuid called with valid leaves
+        && unsafe { std::arch::x86_64::__cpuid(0x8000_0000).eax >= 0x8000_0006 }
+    {
```

Comment:
Don't you need to delete the old CpuIdEntry here? It is got from KVM_GET_SUPPORTED_CPUID.

Reply:
It's intentional. I want to use the version that KVM provides if it has filled it in and only add a new entry if it wasn't provided by KVM.

Reply:
But now the problem is that kvm provides it, but it is wrong.

```
function: 0x80000006, index: 0, flags: 0
kvm support cpuid: eax: 0, ebx: 0, ecx: 0, edx: 0
              host cpuid: eax: 0, ebx: 0, ecx: 1006040, edx: 0
```
In my host, KVM provides cpuid about 0x80000006 that is all reset to 0.

Reply:
Okay, i've updated the PR.

Reply:
May it be better to put it in this branch?
```rust
    // Update some existing CPUID
    for entry in cpuid.as_mut_slice().iter_mut() {
        match entry.function {
```

Reply:
> May it be better to put it in this branch?
> 
> ```rust
>     // Update some existing CPUID
>     for entry in cpuid.as_mut_slice().iter_mut() {
>         match entry.function {
> ```

What a good idea; updated.